pkg/apis/workflow/v1alpha1/task_set_types.go

```diff
@@ -0,0 +1,43 @@
+package v1alpha1
+
+import metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+// +genclient
+// +kubebuilder:subresource:status
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+type WorkflowTaskSet struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata" protobuf:"bytes,1,opt,name=metadata"`
+	Spec              WorkflowTaskSetSpec    `json:"spec" protobuf:"bytes,2,opt,name=spec"`
+	Status            *WorkflowTaskSetStatus `json:"status,omitempty" protobuf:"bytes,3,opt,name=status"`
+}
+
+type WorkflowTaskSetSpec struct {
+	Tasks []Task `json:"tasks,omitempty" protobuf:"bytes,1,rep,name=tasks"`
+}
+
+type Task struct {
+	NodeID   string   `json:"nodeId" protobuf:"bytes,1,opt,name=nodeId"`
+	Template Template `json:"template" protobuf:"bytes,2,opt,name=template"`
```

Comment:
One thought here, if we have many tasks, but they all use the same template, could we re-use the template?